src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -55,169 +59,158 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
-            // `Fully qualify 'Dns' -> System.Net.Dns`
-            else
-            {
-                var expectedCodeActionPrefix = $"Fully qualify '{associatedValue}' -> ";
-                if (codeAction.Title.StartsWith(expectedCodeActionPrefix, StringComparison.OrdinalIgnoreCase))
-                {
-                    fqn = codeAction.Title.Substring(expectedCodeActionPrefix.Length);
+                    var addUsingCodeAction = AddUsingsCodeActionProviderFactory.CreateAddUsingCodeAction(fqn, context.Request.TextDocument.Uri);
+                    if (addUsingCodeAction != null)
+                    {
+                        typeAccessibilityCodeActions.Add(addUsingCodeAction);
+                    }
                 }
             }
 
-            if (string.IsNullOrEmpty(fqn))
-            {
-                typeAccessibilityCodeActions = default;
-                return false;
-            }
-
-            typeAccessibilityCodeActions = new List<RazorCodeAction>();
-
-            var fqnCodeAction = CreateFQNCodeAction(context, diagnostic, codeAction, fqn);
-            typeAccessibilityCodeActions.Add(fqnCodeAction);
-
-            var addUsingCodeAction = AddUsingsCodeActionProviderFactory.CreateAddUsingCodeAction(fqn, context.Request.TextDocument.Uri);
-            if (addUsingCodeAction != null)
-            {
-                typeAccessibilityCodeActions.Add(addUsingCodeAction);
-            }
-
-            return true;
+            return typeAccessibilityCodeActions;
         }
 
-        private static bool TryProcessCodeActionVS(
+        private static IEnumerable<RazorCodeAction> ProcessCodeActionsVS(
             RazorCodeActionContext context,
-            RazorCodeAction codeAction,
-            Diagnostic diagnostic,
-            out ICollection<RazorCodeAction> typeAccessibilityCodeActions)
+            IEnumerable<RazorCodeAction> codeActions)
         {
-            typeAccessibilityCodeActions = new List<RazorCodeAction>(1);
+            var typeAccessibilityCodeActions = new List<RazorCodeAction>(1);
 
-            if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
+            foreach (var codeAction in codeActions)
             {
-                if (codeAction.Children?.Length == 0)
+                if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
                 {
-                    // When there's only one FQN suggestion, code action title is of the form:
-                    // `System.Net.Dns`
-                    var fqn = codeAction.Title;
-                    typeAccessibilityCodeActions.Add(CreateFQNCodeAction(context, diagnostic, codeAction, fqn));
+                    // The formatting pass of our Default code action resolver rejects
+                    // implicit/explicit expressions. So if we're in an implicit/explicit expression, 
```

Comment:
Can we be sure that the edit that we're getting back from C# isn't going to destroy the Razor doc? I feel like this is concerning just letting the edit shine through; although if you're confident here that it would never make sense for edits like these "TO" destroy the document then I can be convinced otherwise.

Reply:
Updated to only support non-formatted edit resolution for **implicit** expressions. I haven't run into any situation, nor can I think of a scenario where an unformatted implicit expression edit could lead to document corruption. As for explicit expressions this is trickier and I can't make as strong of an assertion. Doing some more testing, I ran into potential issues with explicit expressions, which I believe spans from our long-standing issues with respect to document mapping in code actions. Consequently, we now don't offer the FullyQualify code action when in an explicit expression.